### ISSUE TITLE
Unwanted wrapping of link assignments

### ISSUE THREAD

**Issue by jnqnfe** [rust-lang/rustfmt#4933]
I've deliberately defined each link on individual lines, and I've deliberately ignored the typical max line length. rustfmt chooses to not ignore the max line length (debatably not so good) and worse, groups them all up into a single "paragraph".

```diff
-//! [`Introspector::get_sink_info_by_name()`]: crate::context::introspect::Introspector::get_sink_info_by_name
-//! [`SinkFlagSet::DECIBEL_VOLUME`]: crate::def::SinkFlagSet::DECIBEL_VOLUME
-//! [`SourceFlagSet::DECIBEL_VOLUME`]: crate::def::SourceFlagSet::DECIBEL_VOLUME
+//! [`Introspector::get_sink_info_by_name()`]:
+//! crate::context::introspect::Introspector::get_sink_info_by_name [`SinkFlagSet::DECIBEL_VOLUME`]:
+//! crate::def::SinkFlagSet::DECIBEL_VOLUME [`SourceFlagSet::DECIBEL_VOLUME`]:
+//! crate::def::SourceFlagSet::DECIBEL_VOLUME
```

Version string: **rustfmt 1.4.37-stable (a178d03 2021-07-26)**
Project: https://github.com/jnqnfe/pulse-binding-rust
Specific file: https://github.com/jnqnfe/pulse-binding-rust/blob/master/pulse-binding/src/volume.rs
Ran against the master branch.

**Comment by ytmimi** [comment]
@calebcartwright can we close this? I think it was fixed by the merged PR!

**Comment by calebcartwright** [comment]
> @calebcartwright can we close this? I think it was fixed by the merged PR!

Indeed, do you have access to close?

**Comment by ytmimi** [comment]
closing as this was fixed by #5096
